string timestamp
float64[] err